### PR TITLE
Added module - Project Contracts

### PR DESCRIPTION
**Pull request by Kirannegi02**
Targets #2252 

### Description
- Added a new module - Project Contract

### Checklist:

- [x] I have performed a self-review of my own code.

### CODE REVIEW

File: Modules/ProjectContract/Entities/ProjectContractMeta.php [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+{
+    use HasFactory, SoftDeletes;
+
+    protected $table = 'project_contract_meta';
+
+    protected $casts = [
+        'attributes' => 'array',
+    ];
+
+    protected $guarded = [];
```
Comment (gautam071) [inline Modules/ProjectContract/Entities/ProjectContractMeta.php:19]: I think fillable array is missing

File: Modules/ProjectContract/Http/Controllers/ProjectContractController.php [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+            'client_id' => $clientId,
+            'website_url' => $validated['website_url'],
+            'logo_img' => $validated['logo_img'],
+            'authority_name' => $validated['authority_name'],
+            'contract_date_for_signing' => $validated['contract_date_for_signing'],
+            'contract_date_for_effective' => $validated['contract_date_for_effective'],
+            'contract_expiry_date' => $validated['contract_expiry_date'],
+            'attributes' => json_encode($validated['attributes']),
+        ]);
+
```
Comment (gautam071) [inline Modules/ProjectContract/Http/Controllers/ProjectContractController.php:62]: Can we use services file?

File: Modules/ProjectContract/Resources/views/create.blade.php [hunk 3]
```diff
@@ -0,0 +1,21 @@
+@extends('project::layouts.master')
+@section('content')
+
+<div class="container">
+	{{-- @include('project::menu_header') --}}
+    <br><h4>Add new Client</h4>
+    <div class="">
```
Comment (gautam071) [inline Modules/ProjectContract/Resources/views/create.blade.php:7]: why empty class

File: Modules/ProjectContract/Resources/views/index.blade.php [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    </div>
+    <div class="mb-2 mt-2">
+        <form class="d-md-flex justify-content-between ml-md-3">
+            <div class='d-flex justify-content-between align-items-md-center mb-2 mb-xl-0'>
+                <h4 class="">Project Contracts</h4>
+            </div>
+            <div class="d-flex align-items-center">
+                <input type="text" name="name" class="form-control" id="name" placeholder="Client name"
+                value={{request()->get('name')}}>
+                <button class="btn btn-info ml-2 text-white">Search</button>
```
Comment (gautam071) [inline Modules/ProjectContract/Resources/views/index.blade.php:26]: Is this a search button?

File: Modules/ProjectContract/Resources/views/index.blade.php [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+            </div>
+        </form>
+    </div>
+    <div class='d-md-none mb-2'>
+        <div class="d-flex flex-row-reverse">
+            <a href= "{{ route('projectcontract.create') }}" class="btn btn-info text-white">{{ __('Add new project contract') }}</a>
+        </div>
```
Comment (gautam071) [inline Modules/ProjectContract/Resources/views/index.blade.php:33]: Repeated code?

File: Modules/ProjectContract/Resources/views/index.blade.php [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+
+                    @foreach($client->linkedAsPartner as $partnerClient)
+                        @include('client::subviews.listing-client-row', ['client' => $partnerClient, 'level' => 1])
+                    @endforeach
+
+                    @foreach($client->linkedAsDepartment as $department)
+                        @include('client::subviews.listing-client-row', ['client' => $department, 'level' => 1])
+                    @endforeach
+
+                @empty
```
Comment (gautam071) [inline Modules/ProjectContract/Resources/views/index.blade.php:56]: What is this for?

File: database/migrations/2018_03_30_132837_create_projects_table.php [hunk 7]
```diff
@@ -53,6 +53,7 @@ public function down()
             ]);
         });
 
-        Schema::dropIfExists('projects');
+        Schema::dropIfExists('projects
+        ');
```
Comment (gautam071) [inline database/migrations/2018_03_30_132837_create_projects_table.php:57]: Revert back

File: resources/views/layouts/navbar.blade.php [hunk 8]
```diff
@@ -37,24 +37,30 @@
...
+            <div class="dropdown-menu" aria-labelledby="navbarDropdown_finance">
+                @can('clients.view')
+                    <a class="dropdown-item" href="{{ route('client.index') }}">Clients</a>
+                @endcan
+                @can('projects.view')
+                    <a class="dropdown-item" href="{{ route('project.index') }}">Projects</a>
+                @endcan
+                @can('project_contract.view')
+                    <a class="dropdown-item" href="{{ route('projectcontract.index') }}">Project Contract</a>
+                @endcan
```
Comment (gautam071) [inline resources/views/layouts/navbar.blade.php:58]: Have you created seeders as well?

File: resources/views/status.blade.php [hunk 9]
```diff
@@ -2,9 +2,9 @@
 	<div class="alert alert-danger" role="alert">
 		<p><strong>There were some errors. Please resolve them and try again.</strong></p>
 	    <ul>
-	    @foreach ($errors as $message)
-	        <li>{{ $message }}</li>
-	    @endforeach
+			@foreach ($errors as $message)
+				<li>{{ $message }}</li>
+			@endforeach
```
Comment (gautam071) [inline resources/views/status.blade.php:7]: Formatting?

**Review by gautam071 — CHANGES_REQUESTED** [review]
Left some comments

**Comment by gautam071** [comment]
Please resolve the conflicts as well

File: Modules/ProjectContract/Resources/views/index.blade.php (hunk 4 above)
Comment (Kirannegi02) [reply Modules/ProjectContract/Resources/views/index.blade.php:26]: Yes, it is a search button.

File: Modules/ProjectContract/Resources/views/index.blade.php (hunk 6 above)
Comment (Kirannegi02) [reply Modules/ProjectContract/Resources/views/index.blade.php:56]: For adding clients in the Index page.

File: resources/views/layouts/navbar.blade.php (hunk 8 above)
Comment (Kirannegi02) [reply resources/views/layouts/navbar.blade.php:58]: No, I haven't created.

**Pull request closed without merging**